libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -338,12 +390,230 @@ def _search_store(
+                        raise ValueError(
+                            f"Invalid filter condition. Expected $and or $or "
+                            f"but got: {key}"
+                        )
+                else:
+                    # Then it's a field
+                    return self._handle_field_filter(key, filters[key])
+
+                # Here we handle the $and and $or operators
+                if not isinstance(value, list):
+                    raise ValueError(
+                        f"Expected a list, but got {type(value)} for value: {value}"
+                    )
+                if key.lower() == "$and":
+                    and_ = [self._create_filter_clause(el) for el in value]
+                    if len(and_) > 1:
+                        return sqlalchemy.and_(*and_)
+                    elif len(and_) == 1:
+                        return and_[0]
+                    else:
+                        raise ValueError(
+                            "Invalid filter condition. Expected a dictionary "
+                            "but got an empty dictionary"
+                        )
+                elif key.lower() == "$or":
+                    or_ = [self._create_filter_clause(el) for el in value]
+                    if len(or_) > 1:
+                        return sqlalchemy.or_(*or_)
+                    elif len(or_) == 1:
+                        return or_[0]
+                    else:
+                        raise ValueError(
+                            "Invalid filter condition. Expected a dictionary "
+                            "but got an empty dictionary"
+                        )
+
+                else:
+                    raise ValueError(
+                        f"Invalid filter condition. Expected $and or $or "
+                        f"but got: {key}"
+                    )
+
+            elif len(filters) > 1:
+                # Then all keys have to be fields (they cannot be operators)
+                for key in filters.keys():
+                    if key.startswith("$"):
+                        raise ValueError(
+                            f"Invalid filter condition. Expected a field but got: {key}"
+                        )
+                # These should all be fields and combined using an $and operator
+                and_ = [self._handle_field_filter(k, v) for k, v in filters.items()]
+                if len(and_) > 1:
+                    return sqlalchemy.and_(*and_)
+                elif len(and_) == 1:
+                    return and_[0]
+                else:
+                    raise ValueError(
+                        "Invalid filter condition. Expected a dictionary "
+                        "but got an empty dictionary"
+                    )
+            else:
+                raise ValueError("Got an empty dictionary for filters.")
+        else:
+            logging.info("No filters are passed, returning")
+            return None
+
+    def _handle_field_filter(
```

Comment:
would be useful to provide an example input as part of the function description

Reply:
For the filter: {"$or": [{"id": 1}, {"name": "bob"}]}

**__handle_field_filter will result:_**

JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_1) = :JSON_VALUE_2
JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_1) = :JSON_VALUE_2

**_Final filter_clause will result:_**

JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_1) = :JSON_VALUE_2 OR JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_3) = :JSON_VALUE_4

Reply:
Not sure if this should be included in the description as this looks more like objects.

Reply:
Will add it in the description comment

Reply:
if not the exact object format, a description of how the result looks like will be useful to decipher what to expect from the function..

Reply:
Yes, I will add it

Reply:
Added

Reply:
why does the example contain OR? This is field filter so mention only id: 1

Reply:
I did not get you Aniruddh. We have the filetr as, {"$or": [{"id": 1}, {"name": "bob"}]},

**___handle_field_filter will result_**  is called 2 times, {"id": 1} and  {"name": "bob"}. 
Hence, JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_1) = :JSON_VALUE_2
JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_1) = :JSON_VALUE_2


**_create_ filter_clause_** will club these 2 filters with the OR operator, so it will result

JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_1) = :JSON_VALUE_2 OR JSON_VALUE(langchain_vector_store_tests.content_metadata, :JSON_VALUE_3) = :JSON_VALUE_4

Reply:
Updated as discussed